File: features/auth/dev-accounts.ts
// Dev-only convenience — see app/(auth)/phone.tsx's `isDevLoginEnabled`,
// which renders these buttons when EITHER `__DEV__` is true (Expo Go / a
// dev-client on Metro) OR `EXPO_PUBLIC_ENABLE_DEV_LOGIN` is "true" (set in
// eas.json's `development` and `preview` build profiles only — deliberately
// absent from `production`, so this cannot ship to the Play Store no matter
// what).
//
// These are throwaway Supabase Auth accounts with no real customer data
// behind them — they exist so testing sign-in doesn't require a real phone
// number and a real (paid) Twilio SMS send on every single run. Create them
// once via Supabase Dashboard > Authentication > Users > Add user (tick
// "Auto Confirm User"), then give each one a matching `profiles` row:
//
//   insert into public.profiles (id, phone, full_name, role)
//   values ('<uuid from the Users page>', null, 'Dev Customer', 'customer');
//
// Change these passwords to whatever you used when creating the accounts —
// they only unlock an empty test account, never anything real, so there's no
// harm in them living in source.
export const DEV_ACCOUNTS = [
  { label: 'Dev: Customer', email: 'dev-customer@glide.test', password: 'dev-testing-only' },
  { label: 'Dev: Partner', email: 'dev-partner@glide.test', password: 'dev-testing-only' },
] as const;

File: android-native/app/src/main/java/com/glide/app/presentation/auth/DevAccounts.kt
package com.glide.app.presentation.auth

/**
 * Dev-only convenience — mirrors features/auth/dev-accounts.ts exactly (same
 * accounts, same backend). These are throwaway Supabase Auth accounts with no
 * real customer data behind them, so sign-in testing doesn't burn a real
 * (paid) Twilio SMS send on every run.
 *
 * Rendered only when BuildConfig.DEBUG is true (see PhoneEntryScreen) — never
 * reachable in a release build, by construction, not by discipline.
 *
 * dev-admin@glide.test does not exist yet: Supabase's signup validation
 * rejects the reserved ".test" TLD for *new* accounts (the existing
 * customer/partner ones were created earlier via the Dashboard's "Add user",
 * which doesn't apply that check). Create it the same way as the original two
 * — Dashboard > Authentication > Users > Add user (tick "Auto Confirm User"),
 * then: insert into public.profiles (id, full_name, role) values
 * ('<uuid>', 'Dev Admin', 'admin');
 */
data class DevAccount(val label: String, val email: String, val password: String)

val DEV_ACCOUNTS = listOf(
    DevAccount("Dev: Customer", "dev-customer@glide.test", "dev-testing-only"),
    DevAccount("Dev: Partner", "dev-partner@glide.test", "dev-testing-only"),
    DevAccount("Dev: Admin", "dev-admin@glide.test", "dev-testing-only"),
)
